File: ios/CameraViewManager.m
//
//  CameraViewManager.m
//  AndroidModuleTest4
//
//  Created by gloovir on 02/04/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(CameraViewManager, RCTViewManager)
RCT_EXTERN_METHOD(changeCamera:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(record:(id)view)

@end

File: ios/CameraViewManager.swift
//
//  CameraViewManager.swift
//  AndroidModuleTest4
//
//  Created by gloovir on 02/04/2019.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

@available(iOS 10.0, *)
@objc(CameraViewManager)
class CameraViewManager: RCTViewManager{
  let cameraHeimdall = CameraHeimdall()
  override func view() -> UIView! {   
    return cameraHeimdall
    //return CameraHeimdall()
  }
  
  @objc
  func changeCamera(_ node:NSNumber){
    self.cameraHeimdall.changeCamera()
  }
  
  @objc
  func record(_ node:NSNumber, orientation: NSString){
    cameraHeimdall.record(orientation: orientation)
    
  }
}
